res/res_pjsip_logger.c

```diff
@@ -665,15 +668,27 @@ static void check_debug(void)
 	RAII_VAR(char *, debug, ast_sip_get_debug(), ast_free);
 
 	if (ast_false(debug)) {
-		pjsip_disable_logger(-1);
+		/* If the logger was enabled via the CLI instead of
+		 * through the config file, then we shouldn't
+		 * disable it on a reload.
+		 * Only disable logging if the logger was previously
+		 * enabled via the config file. */
+		if (logger_enabled_via_config) {
+			pjsip_disable_logger(-1);
+		}
+		logger_enabled_via_config = 0;
 		return;
 	}
 
 	if (ast_true(debug)) {
+		if (logger_enabled_via_config) {
+			return;
+		}
 		pjsip_enable_logger_all(-1);
 		return;
 	}
 
+	logger_enabled_via_config = 1;
```

Comment:
This doesn't behave as described. The logger_enabled_via_config is only set when host based logging is enabled from the configuration file. If you merely enable it globally for all traffic, it's never set.